client/src/pages/NewNote/NewNote.tsx

```diff
@@ -0,0 +1,96 @@
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
+import { Button } from "@heroui/react";
+import { FilePlus2 } from "lucide-react";
+
+import { addNotes, selectAuthUser } from "@/features";
+import { useAppDispatch, useAppSelector } from "@/hooks";
+import { supabase } from "@/supabase";
+
+import { NewNoteFormData } from "./NewNote.types";
+
+export const NewNotePage = () => {
+  const dispatch = useAppDispatch();
+  const user = useAppSelector(selectAuthUser);
+
+  const navigate = useNavigate();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { isSubmitting },
+  } = useForm<NewNoteFormData>();
+
+  const onSubmit: SubmitHandler<NewNoteFormData> = async (data) => {
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    const { data: notes, error } = await supabase
+      .from("note")
+      .insert([
+        {
+          title: data.title,
+          notes: data.notes,
+          in_trash: false,
+          user_id: user.id,
+        },
+      ])
+      .select();
+
+    if (error) {
+      console.error(error);
+      return;
+    } else {
+      dispatch(addNotes(notes));
+      navigate(`/note/${notes[0]}`);
+    }
+  };
+
+  return (
+    <div className="max-w-app mx-auto pt-5">
+      <div className="px-2">
+        <h1 className="text-center text-3xl font-semibold">Create Note</h1>
+
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <div className="mt-10 flex flex-col gap-4">
+            <div className="my-8">
+              <input
+                type="text"
+                autoComplete="off"
+                disabled={isSubmitting}
+                {...register("title", {
+                  required: true,
+                  validate: (value) => {
+                    if (value.length === 0) {
+                      return "Title is required";
+                    }
+                  },
```

Comment:
This error message is not used anywhere. Check and remove if needed. If needed, add an error message in the form